### PR TITLE
Create all interests and people

### LINKED ISSUE #2: Create a list of interests

### LINKED ISSUE #3: Generate a number of participants

### LINKED ISSUE #4: Populate the initial data container (participants with their interests)

### LINKED ISSUE #5: Log participants’ introductions

### PR DESCRIPTION
**Pull request by melnelen**
This ticket
resolves #2 
resolves #3 
resolves #4 
resolves #5.

### CODE REVIEW

File: CuriousKatie/Helper/Helper.swift [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+//  Helper.swift
+//  CuriousKatie
+//
+//  Created by Alexandra Ivanova on 30/04/2020.
+//  Copyright © 2020 Alex Ivanova. All rights reserved.
+//
+
+import Foundation
+
+struct Helper {
```
Comment (cyril-lepottier) [inline CuriousKatie/Helper/Helper.swift:11]: Alternatively you can remove the struct and have all the methods and properties as global functions and constants.

File: CuriousKatie/Helper/Helper.swift [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+                person09,
+                person10,
+                person11,
+                person12]
+    }
+    
+    /// Shuffle interests and pick a random amount between 1 and the maximum number of interests per person
+    /// *Note: in case the number of interests is smaller than maximum number of interests per person return at least the available;e number of interests
+    static func pickSomeInterests() -> [Interest] {
+        let pool = generateAllInterests().shuffled()
```
Comment (cyril-lepottier) [inline CuriousKatie/Helper/Helper.swift]: You can have a slight performance gain by only generating the interests once.

File: CuriousKatie/ViewController.swift [hunk 3]
```diff
@@ -9,10 +9,24 @@
...
+        for person in listOfPeople {
+            print(person.introduceMyself())
+        }
+        
+        /// Print a new line to separate sections.
+        print("\n")
+        
+        /// All participants share all of their interests.
+        for person in listOfPeople {
+            print(person.shareMyInterests())
```
Comment (cyril-lepottier) [inline CuriousKatie/ViewController.swift:29]: Are you not supposed to log **one** unshared interest **at a time**?

File: CuriousKatie/Helper/Helper.swift [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+                interest31,
+                interest32,
+                interest33,
+                interest34,
+                interest35,
+                interest36]
+    }
+    
+    /// Generate all the people
+    static func generateAllPeople() -> [Person] {
```
Comment (cyril-lepottier) [inline CuriousKatie/Helper/Helper.swift:62]: This generator does't ensure the unicity of the name, does it?

**Review by cyril-lepottier — CHANGES_REQUESTED** [review]
There is 2 potential bugs that I think need to be fixed.

File: CuriousKatie/ViewController.swift (hunk 3 above)
Comment (cyril-lepottier) [reply CuriousKatie/ViewController.swift:29]: I just realized that you were planing on implementing this later as part of #6, you can ignore the previous comment.

File: CuriousKatie/Helper/Helper.swift (hunk 4 above)
Comment (melnelen) [reply CuriousKatie/Helper/Helper.swift:62]: There is no function to check that the names are unique... I can check for that, although with the number of different combinations that are possible the chance of having even two identical names is probably 0.0000000000000000001 😉

File: CuriousKatie/Helper/Helper.swift (hunk 2 above)
Comment (melnelen) [reply CuriousKatie/Helper/Helper.swift]: Here I'm trying to pick a random number and random interests for each person. I didn't have any other idea on how to do it 😐

File: CuriousKatie/Helper/Helper.swift (hunk 1 above)
Comment (melnelen) [reply CuriousKatie/Helper/Helper.swift:11]: I guess I could. Is it a better approach?

File: CuriousKatie/Helper/Helper.swift (hunk 2 above)
Comment (melnelen) [reply CuriousKatie/Helper/Helper.swift]: Fixed 😎

File: CuriousKatie/Helper/Helper.swift (hunk 4 above)
Comment (melnelen) [reply CuriousKatie/Helper/Helper.swift:62]: Fixed 😎

File: CuriousKatie/ViewController.swift (hunk 3 above)
Comment (melnelen) [reply CuriousKatie/ViewController.swift:29]: Yes, ok 🙃This was just to check that I'm actually creating random interests for each person.

File: CuriousKatie/Helper/Helper.swift (hunk 1 above)
Comment (cyril-lepottier) [reply CuriousKatie/Helper/Helper.swift:11]: No, just an alternative one FYI.

File: CuriousKatie/Helper/Helper.swift [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+                interest27,
+                interest28,
+                interest29,
+                interest30,
+                interest31,
+                interest32,
+                interest33,
+                interest34,
+                interest35,
+                interest36]
```
Comment (cyril-lepottier) [inline CuriousKatie/Helper/Helper.swift]: Just saw in you last commit (184f3d9f47344c6ba20be168da92e02f2775f130) that adding some interests made you rename a bunch of variables. An easy improvement is to simply remove the variables:
```
return [
   Interest(name: "acting", category: Category.art, gear: [Gear.noGear], environment: Environment.anywhere),
   Interest(name: "aerial acrobatics", category: Category.sport, gear: [Gear.sportClothes], environment: Environment.indoor),
   Interest(name: "astronomy", category: Category.science, gear: [Gear.optical, Gear.photo], environment: Environment.outdoor),
   Interest(name: "bike touring", category: Category.sport, gear: [Gear.ecoTransport], environment: Environment.outdoor),
   ...
]
```

**Review by cyril-lepottier — APPROVED** [review]

File: CuriousKatie/Helper/Helper.swift (hunk 5 above)
Comment (melnelen) [reply CuriousKatie/Helper/Helper.swift]: Done 😋

**Pull request merged**